pkg/route/graph/analysis/analysis.go

```diff
@@ -70,6 +75,54 @@ route:
 
 				continue route
 			}
+
+			if routeNode.Spec.Port != nil {
+				routePortString := routeNode.Spec.Port.TargetPort.String()
+				if routePort, err := strconv.Atoi(routePortString); err == nil {
+					// route has numeric port, service has no port with that number as a targetPort.
+					for _, port := range svcNode.Spec.Ports {
+						if port.TargetPort.IntValue() == routePort {
+							continue route
```

Comment:
Comment in the same fashion as https://github.com/openshift/origin/pull/7833#discussion_r55297690. We could break these down to separate self-contained units but continuing has served us well so far. If we decide that it's a real problem, we should refactor all marking functions.